File: clientes/fryda-cafe-taubate/apresentacao-site/gerar_pdf.py
#!/usr/bin/env python3
"""
Gera os PDFs (apresentação + proposta) da versão "site grátis" via WeasyPrint.

Use este na nuvem / Claude Code na web, onde o Playwright/Chromium não baixa.
Localmente, o `gerar-pdf.js` (Playwright) dá a maior fidelidade. A fonte de
verdade são sempre os HTMLs.

Uso:
    python gerar_pdf.py            # gera os dois
    python gerar_pdf.py apresentacao.html Apresentacao-Fryda-Site.pdf

Requer: pip install weasyprint
"""

import sys
import warnings
from pathlib import Path

try:
    from weasyprint import HTML
except ImportError:
    sys.exit("WeasyPrint não instalado. Rode: pip install weasyprint")

base = Path(__file__).resolve().parent
ALVOS = [
    ("apresentacao.html", "Apresentacao-Fryda-Site.pdf"),
    ("proposta.html", "Proposta-Fryda-Site.pdf"),
]

warnings.filterwarnings("ignore")
args = sys.argv[1:]
if len(args) >= 2:
    ALVOS = [(args[0], args[1])]

for html_name, pdf_name in ALVOS:
    html = base / html_name
    if not html.exists():
        sys.exit(f"HTML não encontrado: {html}")
    HTML(filename=str(html), base_url=str(base)).write_pdf(str(base / pdf_name))
    print(f"PDF gerado: {pdf_name}")

File: clientes/fryda-cafe-taubate/apresentacao-so-site/gerar-pdf.js
// Gera os PDFs (apresentação + proposta) da versão "só site" (sem GMB) via Chromium (Playwright).
// Uso: NODE_PATH="<pasta-com-node_modules>/node_modules" node gerar-pdf.js
const { chromium } = require('playwright');
const path = require('path');

const ALVOS = [
  ['apresentacao.html', 'Apresentacao-Fryda-Site.pdf'],
  ['proposta.html', 'Proposta-Fryda-Site.pdf'],
];

(async () => {
  const browser = await chromium.launch();
  for (const [html, pdf] of ALVOS) {
    const page = await browser.newPage();
    const file = 'file://' + path.resolve(__dirname, html).replace(/\\/g, '/');
    await page.goto(file, { waitUntil: 'networkidle' });
    await page.evaluate(() => document.fonts.ready);
    await page.waitForTimeout(300);
    await page.pdf({
      path: path.join(__dirname, pdf),
      format: 'A4',
      printBackground: true,
      preferCSSPageSize: true,
    });
    await page.close();
    console.log('PDF gerado: ' + pdf);
  }
  await browser.close();
})();
